### DISCUSSION TITLE
How can I deploy a Django app which uses an external database?

### DISCUSSION THREAD

**Discussion by jgtvares** [vercel/vercel#7236] [Help] [answered]
Hey everyone!
I have an API that I developed using [Django Rest Framework](https://www.django-rest-framework.org/) and I realized Vercel does not have a runtime for this use case.
Doing some research I stumbled in this [example](https://github.com/jayhale/vercel-django-example).
But the problem is that it is not compatible for Django apps that use an external database, which is my case.
When I deploy my API the build fails when trying to install `mysqlclient` Python package.
I've tested removing this dependency and the build passed, but the API does not work properly - obviously.

So here are my questions:
1. Is there a known workaround for this use case?
2. Is this use case possible if I try to develop my own runtime?
3. Is there another Python framework which would be more plug and play (with external database support)?

**Comment by ghost** [comment]
What error do you get? Maybe this helps? https://stackoverflow.com/questions/35763380/problems-using-mysql-with-aws-lambda-in-python

**Comment by jgtvares** [comment]
@roelandmoors thank you for answering!
I've managed to make the deployment work. I've found a custom Python runtime, forked it, and made some changes.
This is the repo containing my version: https://github.com/jgtvares/py-vercel
Deploy is working fine, but now Django is throwing a generic error, which I can't seem to debug. Check out the preview [link](https://hackz-trigger-api-cg96ij94b-hackzcompany.vercel.app/).
Searching around for this, some people claim that I can edit an internal Django file to show me the exact error, but I can't edit this in Vercel environment.

**Comment by jgtvares** [comment, marked as answer]
@roelandmoors 
I'm glad to say it is working properly now!
I forked Django itself and made the internal lib changes, then I was able to see the actual error, which was: `No module named _mysql`.
Searching around it seems to be some lack of dependencies in the Lambda instance, but I already added a step to install `mysql-devel` and it made `mysqlclient` Python package installation possible. So I didn't have anywhere else to go from there.
After trying a lot of things, I decided to change from `mysqlclient` to `pymysql` and after two lines of changes in my project I can finally access the endpoints in my Vercel deployment.

So, answering my own question, if you want to deploy a Django app using MySQL libs you have to use `pymysql` to avoid any headaches.
My [custom runtime](https://github.com/jgtvares/py-vercel) will be available, you can use it.

### Disclaimer
It is probably possible to use `mysqlclient`, in my custom runtime I was able to install system dependencies in the Lambda instance in order to make Python database libs installable, that's some progress.
But the difference here is that Vercel uses WSGI to make the app accessible and that's a different approach from the `manage.py runserver` command. So if you have time to invest in that line of investigation you might get there - please, let me know if you do.

**Comment by kumard3** [comment]
does anyone have any working example of Django with postgres on vercel

**Reply by HackyRoot** [reply]
@kumard3 someone created a tutorial around Django with Postgres on vercel: https://omavhad.hashnode.dev/deploying-django-with-postgresql-on-vercel.
